app/views/producers/index.html.haml

```diff
@@ -2,7 +2,8 @@
   = t :producers_title
 
 - content_for :injection_data do
-  = inject_enterprises(@enterprises)
+  - cache @enterprises do
```

Comment:
Hm, I thought that this may fail when enterprise relationships change ("Shop for ... products at") but EnterpriseRelationship touches the enterprises and the updated timestamps should invalidate this cache, right?

Reply:
> "Shop for ... products at"

You mean when you click on an enterprise and it pulls down to reveal the tab that lists the hubs for a producer?

That data is loaded asynchronously via an API call when the tab is first clicked, so none of what's shown inside the tabs is being cached here. We're basically just caching the data that populates the initial list of producers on the page.

But yes, there's a touch there too which would update the enterprise if it's relationships change.

Reply:
For reference; we can't use the same technique on the `/shops` page (even though they're very similar), as the data there is heavily dependent on whether or not each shop's order cycles are currently open or closed (which is really difficult to work in to a caching strategy). On this page all producers are shown without that condition.